scylla-server/src/proxy/proxy-client.ts

```diff
@@ -39,26 +61,108 @@ export default class ProxyClient {
    */
   private handleOpen = (packet: IConnackPacket) => {
     console.log('Connected to Siren', packet.properties);
+    this.createNewRun = true;
     this.subscribeToTopics(Object.values(Topic));
   };
 
   /**
    * Handles messages received from Siren
+   * Parses as ServerMessage, otherwise throws error
+   * Passes data to handleData
    * @param topic The topic the message was received on
    * @param message The message received from Siren
    */
-  private handleMessage = (topic: string, payload: Buffer) => {};
-
+  private handleMessage = (topic: string, payload: Buffer) => {
+    // am i supposed to do something with topic
+    console.log('Received Message: ', payload);
+    try {
+      const data = JSON.parse(payload.toString()) as ServerMessage;
+      this.handleData(data);
+    } catch (error) {
+      if (error instanceof Error) {
+        console.log('Error Decoding Message: ', error.message);
+        this.handleError(error);
+      }
+    }
+  };
   /**
    * Handles receiving data from the car and:
    * 1. Logs the data
    * 2. Sends the data to the client
    * @param data The data received from Siren
    */
-  private handleData = (data: ServerMessage) => {
-    //TODO: Send data to client
-    //TODO: Log data
-    console.log('Received Data: ', data);
+  private handleData = async (data: ServerMessage) => {
+    // if first time data recieved since connecting to car
+    // then create new run
+    if (this.createNewRun) {
+      this.currentRun = await RunService.createRun(data.unix_time);
+    }
+    this.createNewRun = false;
+    // upsert the node
+    const node = await NodeService.upsertNode(data.node);
+    // start upserting data
+    if (!this.currentRun) {
+      console.log('no current run');
+      return;
+    }
+    // initializing params
+    let latitude = undefined;
+    let longitude = undefined;
+    let radius = undefined;
+    let locationName = undefined;
+    let driverName = undefined;
+    let systemName = undefined;
+
+    // iterating and upserting
+    const clientData: ClientData[] = [];
+    for (const serverdata of data.data) {
+      // transform serverdata into client data to send to ProxyServer
+      const clientdata: ClientData = {
```

Comment:
like right here, we just send all the data to the frontend regardless of if the data type, system, location, etc has been correctly added to the database. I would move this adding to the clientData to the bottom of the for loop